cpp/test/mst.cu

```diff
@@ -14,37 +14,168 @@
+  for(auto v = 1; v < csr_h.n_vertices; v++) {
+    total_weight += curr_edge[v];
+  }
 
-TEST(Raft, MST) {
-  using namespace mst;
-  using vertex_t = int;
-  using edge_t = int;
-  using value_t = float;
+  return total_weight;
 
-  handle_t h;
-  ASSERT_EQ(0, h.get_num_internal_streams());
+}
+
+template <typename vertex_t, typename edge_t, typename value_t>
+class MSTTest : public ::testing::TestWithParam<csr_t<vertex_t, edge_t, value_t>> {
+  protected:
+    void mst_sequential() {
+      csr_h = ::testing::TestWithParam<csr_t<vertex_t, edge_t, value_t>>::GetParam();
+
+      rmm::device_vector<vertex_t> mst_src;
+      rmm::device_vector<vertex_t> mst_dst;
+
+      MST_solver<vertex_t, edge_t, value_t> solver(handle, csr_d.offsets, csr_d.indices, csr_d.weights, csr_d.n_vertices,
+                                                  csr_d.n_edges);
+
+      //nullptr expected to trigger exceptions
+      EXPECT_ANY_THROW(solver.solve(mst_src, mst_dst));
+    }
+
+    void SetUp() override {
+      csr_d.n_vertices = csr_h.n_vertices;
+      csr_d.n_edges = csr_h.n_edges;
+
+      CUDA_CHECK(cudaMalloc(&csr_d.offsets, csr_d.n_vertices * sizeof(vertex_t)));
+      CUDA_CHECK(cudaMalloc(&csr_d.indices, csr_d.n_edges * sizeof(vertex_t)));
+      CUDA_CHECK(cudaMalloc(&csr_d.weights, csr_d.n_edges * sizeof(vertex_t)));
+
+      raft::update_device(csr_d.offsets, csr_h.offsets, csr_h.n_vertices, handle.get_stream());
+      raft::update_device(csr_d.indices, csr_h.indices, csr_h.n_edges, handle.get_stream());
+      raft::update_device(csr_d.weights, csr_h.weights, csr_h.n_edges, handle.get_stream());
+    }
 
-  edge_t* offsets{nullptr};
-  vertex_t* indices{nullptr};
-  value_t* weights{nullptr};
-  edge_t e = 0;
-  vertex_t v = 0;
-  rmm::device_vector<vertex_t> mst_src;
-  rmm::device_vector<vertex_t> mst_dst;
+    void TearDown() override {
+      CUDA_CHECK(cudaFree(csr_d.offsets));
+      CUDA_CHECK(cudaFree(csr_d.indices));
+      CUDA_CHECK(cudaFree(csr_d.weights));
+    }
+  
+  protected:
+    csr_t<vertex_t, edge_t, value_t> csr_h, csr_d;
 
-  MST_solver<vertex_t, edge_t, value_t> solver(h, offsets, indices, weights, v,
-                                               e);
+    raft::handle_t handle;
+};
 
-  //nullptr expected to trigger exceptions
-  EXPECT_ANY_THROW(solver.solve(mst_src, mst_dst));
+/*
+Graph 1:
+    2
+(0) - (1)
+3| 4\ 1|
+(2)   (3)
+
+*/
+constexpr csr_t<int, int, int> graph1() {
+  int offsets[5] = {0, 3, 5, 7, 8};
+  int indices[8] = {1, 2, 3, 0, 3, 0, 0, 1};
+  int weights[8] = {2, 3, 4, 2, 1, 3, 4, 1};
+  return csr_t<int, int, int> {offsets, indices, weights, 4, 8};
+}
+
+const std::vector<csr_t<int, int, int>> csr_in_h = {
+  // {nullptr, nullptr, nullptr, 0, 0},
+  graph1()
+};
+
+typedef MSTTest<int, int, int> MSTTestSequential;
+TEST_P(MSTTestSequential, Sequential) {
+  // this->mst_sequential();
+
+  // do assertions here
+  // in this case, running sequential MST
+  std::cout << prims(csr_h);
 }
+
```

Comment:
You could take a look at https://github.com/rapidsai/cugraph/blob/branch-0.17/cpp/tests/centrality/betweenness_centrality_test.cu to compare to how graph tests are setup in cugraph.

Or if we get the bulk of the testing in python later, we can also consider a simpler design like 
https://github.com/rapidsai/cugraph/blob/60af376ac9d8bf41e987108a8da07a15e25ecbb7/cpp/tests/renumber/renumber_test.cu#L75